### PR TITLE
feat: Text interface rework

### PR DESCRIPTION
**Pull request by TimonGeisbauer**

### CODE REVIEW

File: dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
- * interact with this behavior.
+ * On applicable components, creates enum which helps facilitate underlying BBj constant 
+ * integers for horizontal alignment behavior to legible enum values, and facilitates 
+ * implementation of methods to interact with this behavior.
  */
-public interface TextAlignable {
+public interface HorizontalAlignment {
 
   /**
    * Enum which represents the text alignment of a component.
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java:12]: `value` should be private

File: dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java [hunk 2]
```diff
@@ -35,14 +35,14 @@ public int getValue() {
    *
    * @return Enum value of text alignment
    */
-  Alignment getTextAlignment();
+  Alignment getHorizontalAlignment();
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java:38]: add public modifer

File: dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java [hunk 3]
```diff
@@ -35,14 +35,14 @@ public int getValue() {
...
   /**
-   * Sets the horizontal alignment of the text within the control.
+   * Sets the horizontal alignment of the text within the component.
    *
    * @param alignment Enum from list representing an internal BBj numeric constant
-   * @return The control itself
+   * @return The component itself
    */
-  TextAlignable setTextAlignment(Alignment alignment);
+  HorizontalAlignment setHorizontalAlignment(Alignment alignment);
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java:46]: add public modifer

File: dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * On applicable component, creates enum which helps facilitate underlying BBj constant 
+ * integers for text position behavior to legible enum values, and facilitates 
+ * implementation of methods to interact with this behavior. Text position is 
+ * the location of the text in relation to the component itself.
+ */
+public interface TextPosition {
+
+  /**
+   * Enum which represents the text positon of a component.
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java:12]: Fix typo in `position`

File: dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+   */
+  enum Position {
+    LEFT(8192), RIGHT(32768);
+
+    public final int value;
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java:17]: `value` must private

File: dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
+package org.dwcj.component;
+
+/**
+ * On applicable component, creates enum which helps facilitate underlying BBj constant
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/TextPosition.java:4]: Fix file formatting

File: dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java [hunk 7]
```diff
@@ -2,11 +2,11 @@
 
 
 /**
- * On applicable controls, creates enum which helps facilitate underlying BBj constant integers for
- * text alignment behavior to legible enum values, and facilitates implementation of methods to
- * interact with this behavior.
+ * On applicable components, creates enum which helps facilitate underlying BBj constant
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/HorizontalAlignment.java:5]: Fix formatting

File: dwcj-engine/src/main/java/org/dwcj/component/VerticalAlignment.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public interface VerticalAlignment {
+
+  /**
+   * Enum which represents the text alignment of a component.
+   */
+  enum Alignment {
+    TOP(8192), CENTER(16384), BOTTOM(32768);
+
+    public final int value;
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/VerticalAlignment.java:16]: value must be private

File: dwcj-engine/src/main/java/org/dwcj/component/VerticalAlignment.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
+package org.dwcj.component;
+
+/**
+ * On applicable components, creates enum which helps facilitate underlying BBj constant
```
Comment (hyyan) [inline dwcj-engine/src/main/java/org/dwcj/component/VerticalAlignment.java:4]: Fix document formatting

**Review by hyyan — APPROVED** [review]

**Comment by hyyan** [comment]
@mhawkinsbasis please review the javadocs and make sure to add what we in #339

**Review by mhawkinsbasis — APPROVED** [review]
Updated the Javadocs, looks good now!

**Review by mhawkinsbasis — APPROVED** [review]
Updated the Javadocs, looks good now!

**Pull request merged**